--- a/sympy/geometry/point.py
+++ b/sympy/geometry/point.py
@@ -266,8 +266,10 @@ class Point(GeometryEntity):
         sqrt(x**2 + y**2)
 
         """
-        return sqrt(sum([(a - b)**2 for a, b in zip(
-            self.args, p.args if isinstance(p, Point) else p)]))
+        if not isinstance(p, Point):
+            p = Point(p)
+        if len(p.args) != len(self.args):
+            raise ValueError("Dimension mismatch between {} and {}".format(self, p))
+        return sqrt(sum([(a - b)**2 for a, b in zip(self.args, p.args)]))
 
     def taxicab_distance(self, p):
         """The Taxicab Distance from self to point p.
