--- a/sympy/geometry/point.py
+++ b/sympy/geometry/point.py
@@ -266,8 +266,11 @@ class Point(GeometryEntity):
         sqrt(x**2 + y**2)
 
         """
-        return sqrt(sum([(a - b)**2 for a, b in zip(
-            self.args, p.args if isinstance(p, Point) else p)]))
+        p = p.args if isinstance(p, Point) else p
+        s, p = list(self.args), list(p)
+        n = max(len(s), len(p))
+        s.extend([S.Zero] * (n - len(s)))
+        p.extend([S.Zero] * (n - len(p)))
+        return sqrt(sum([(a - b)**2 for a, b in zip(s, p)]))
 
     def taxicab_distance(self, p):
         """The Taxicab Distance from self to point p.
